quickwit-config/src/index_config.rs

```diff
@@ -19,70 +19,203 @@
 
 use std::ffi::OsStr;
 use std::path::Path;
+use std::time::Duration;
 
 use anyhow::{bail, Context};
 use byte_unit::Byte;
-use quickwit_index_config::FieldMappingEntry;
+use json_comments::StripComments;
+use quickwit_index_config::{FieldMappingEntry, SortBy, SortOrder};
+use quickwit_storage::load_file;
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct DocMapping {
     pub field_mappings: Vec<FieldMappingEntry>,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
     pub tag_fields: Vec<String>,
+    #[serde(default = "DocMapping::default_store_source")]
     pub store_source: bool,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+impl DocMapping {
+    fn default_store_source() -> bool {
+        true
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct IndexingResources {
+    #[serde(default = "IndexingResources::default_num_threads")]
     pub num_threads: usize,
+    #[serde(default = "IndexingResources::default_heap_size")]
     pub heap_size: Byte,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+impl IndexingResources {
+    fn default_num_threads() -> usize {
+        1
+    }
+
+    fn default_heap_size() -> Byte {
+        Byte::from_bytes(2_000_000_000) // 2GB
+    }
+
+    pub fn for_test() -> Self {
+        Self {
+            num_threads: 1,
+            heap_size: Byte::from_bytes(20_000_000), // 20MB
+        }
+    }
+}
+
+impl Default for IndexingResources {
+    fn default() -> Self {
+        Self {
+            num_threads: Self::default_num_threads(),
+            heap_size: Self::default_heap_size(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct MergePolicy {
+    #[serde(default = "MergePolicy::default_demux_factor")]
     pub demux_factor: usize,
+    #[serde(default = "MergePolicy::default_merge_factor")]
     pub merge_factor: usize,
+    #[serde(default = "MergePolicy::default_max_merge_factor")]
     pub max_merge_factor: usize,
-    pub min_level_num_docs: usize,
 }
 
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+impl MergePolicy {
+    fn default_demux_factor() -> usize {
+        8
+    }
+
+    fn default_merge_factor() -> usize {
+        10
+    }
+
+    fn default_max_merge_factor() -> usize {
+        12
+    }
+}
+
+impl Default for MergePolicy {
+    fn default() -> Self {
+        Self {
+            demux_factor: Self::default_demux_factor(),
+            merge_factor: Self::default_merge_factor(),
+            max_merge_factor: Self::default_max_merge_factor(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct IndexingSettings {
-    pub demux_field: String,
-    pub timestamp_field: String,
+    #[serde(default)]
+    pub demux_enabled: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
The comment above about tags does not apply here of course, because this is an advanced feature we don't want to advertise